nominal/nominal.py

```diff
@@ -1,628 +1,316 @@
-                    dft = df.sort(ts_col)  # Nominal datasets must be sorted by their time series index
-                else:
-                    print("Please specify a relative timestamp column with [code]ts_col[/code]")
-                    return
-            else:
-                dft = Ingest.set_ts_index(df)
-
-        super().__init__(dft)
-
-        self.s3_path = None
-        self.filename = filename
-        self.properties = properties
-        self.description = description
-        self.rid = rid
-        self.dataset_link = ""
-        self.ts_col = ts_col
-        self.relative = relative
-        self.relative_units = relative_units
-
-    def __get_headers(self, content_type: str = "json") -> dict:
-        TOKEN = kr.get_password("Nominal API", "python-client")
-        return {
-            "Authorization": "Bearer {}".format(TOKEN),
-            "Content-Type": "application/{0}".format(content_type),
-        }
-
-    def __upload_file(self, overwrite: bool) -> requests.Response:
-        """
-        Uploads dataframe to S3 as a file.
-
-        Returns:
-        Response object from the REST call.
-        """
-
-        if self.s3_path is not None and not overwrite:
-            print(
-                "\nThis Dataset is already uploaded to an S3 bucket:\n{0}\nTry [code]upload(overwrite = True)[/code] to overwrite it.".format(
-                    self.s3_path
-                )
-            )
-            return
-
-        # Create a default dataset name
-        if self.filename is None:
-            self.filename = default_filename("DATASET")
-
-        csv_file_buffer = io.BytesIO()
-        self.write_csv(csv_file_buffer, datetime_format=r"%Y-%m-%dT%H:%M:%S.%fZ")
-
-        # Get the size of the buffer in bytes
-        csv_file_buffer.seek(0, os.SEEK_END)
-        csv_buffer_size_bytes = csv_file_buffer.tell()
-        csv_file_buffer.seek(0)
-
-        print(
-            "\nUploading: [bold green]{0}[/bold green]\nto {1}\n = {2} bytes".format(
-                self.filename, get_base_url(), csv_buffer_size_bytes
-            )
-        )
 
-        # Make POST request to upload data file to S3
-        resp = requests.post(
-            url=ENDPOINTS["file_upload"].format(get_base_url(), self.filename),
-            data=csv_file_buffer.read(),
-            params={"sizeBytes": csv_buffer_size_bytes},
-            headers=self.__get_headers(content_type="octet-stream"),
+def get_default_connection() -> NominalClient:
+    """Retrieve the default global connection to the Nominal platform.
+
+    Raises nominal.exceptions.NominalError if no global connection has been set.
+    """
+    global _default_connection
+    if _default_connection is None:
+        raise NominalError("No default connection set: initialize with `set_default_connection(base_url, token)`")
+    return _default_connection
+
+
+def upload_dataset_from_pandas(
+    df: pd.DataFrame,
+    name: str,
+    description: str,
```

Comment:
technically description is optional -- should I default it to None for this API for fewer required args, or should we push users into being good stewards by making it required?

Reply:
similarly, should `name` be required or should we feel free to randomize it if not provided?

Reply:
name: required unless we can infer it from something (e.g do dataframes have names or keep track of the csv they came from?)
description: optional, if you make it required people just put empty string or "." - if it's not a valuable feature to them, I don't think we should try and force usage

Reply:
> (e.g do dataframes have names or keep track of the csv they came from?)
nope